<?xml version="1.0"?>
<launch>
    <!-- Include main px4ctrl launch file -->
    <include file="$(find px4ctrl)/launch/px4_ctrl.launch" />
    
    <!-- Circular trajectory node -->
    <node pkg="px4ctrl" type="circular_trajectory.py" name="circular_trajectory" output="screen">
        <!-- Trajectory parameters -->
        <param name="radius" value="3.0" />           <!-- Circle radius in meters -->
        <param name="height" value="2.0" />           <!-- Flight height in meters -->
        <param name="angular_velocity" value="0.3" /> <!-- Angular velocity in rad/s -->
        <param name="center_x" value="0.0" />         <!-- Circle center X -->
        <param name="center_y" value="0.0" />         <!-- Circle center Y -->
        
        <!-- Control parameters -->
        <param name="publish_rate" value="50.0" />    <!-- Command publish rate in Hz -->
    </node>
</launch>